sdks/python/apache_beam/runners/portability/fn_api_runner/worker_handlers.py

```diff
@@ -1087,14 +1150,49 @@ def append_raw(
   ):
     # type: (...) -> _Future
     with self._lock:
-      self._state[self._to_key(state_key)].append(data)
+      if state_key.WhichOneof('type') == 'ordered_list_user_state':
+        coder = TupleCoder(
+            [VarIntCoder(), coders.LengthPrefixCoder(BytesCoder())]).get_impl()
+
+        if self._to_key(state_key) not in self._ordered_list_keys:
+          self._ordered_list_keys[self._to_key(state_key)] = SortedSet()
+
+        for key, value in coder.decode_all(data):
```

Comment:
I simplifed the code with an 1-interval helper. 

However, I will not be able to do `dict(coder.decode_all(data))` because the data here may contain multiple entries with the same key (like multi-map)